### DISCUSSION TITLE
Parameterization is not invariant under isometry

### DISCUSSION THREAD

**Discussion by stla** [CGAL/cgal#7740] [Q&A]
Hello,

 I split a ball into two parts, like a tennis ball, and using the same method, I map a checkerboard on each part:

![TennisBallCheckerboard](https://github.com/CGAL/cgal/assets/4466543/42f80fb8-63f1-47a0-93c4-25efd2ef79af)

The two parts are "the same" (they are isometric, and more or less the same number of vertices). However the two mapped checkerboards are quite different. Why that? Here, to map a checkerboard, I used a DCP parameterization with a square border, then I drew a checkboard on the square that I mapped to the half sphere.

Is there a CGAL parameterization which would yield the same checkerboards on the two isometric parts of the sphere?

**Comment by afabri** [comment]
My guess is that the border parameterization puts the 4 "corners"  at different points on the two borders. Equidistant on both, but shifted arbitrarily.

**Comment by stla** [comment]
Yep.
So, in order to get the same checkerboard on the two halves, I "painted" only one half and I constructed the other by rotating the first one (and I changed the colors).

![TennisBallWithCheckerboard_BW](https://github.com/CGAL/cgal/assets/4466543/ea286a97-112e-4943-8c92-58126374049f)

I use the DCP parameterization and however the angles of the checkerboard are not close to right angles. Maybe I should rotate the mesh before computing the parameterization, and this will change the angles?

**Reply by afabri** [reply]
Did you try  [ARAP](https://doc.cgal.org/latest/Surface_mesh_parameterization/index.html#title15)  where you have a parameter for angle/shape?
Concerning the border parameterization the 4 corners should be roughly here: 
![IMG_1666](https://github.com/CGAL/cgal/assets/3263539/b404b681-7e1f-4b3a-ab9c-f92ac2aa5e83)

**Reply by afabri** [reply]
See [this constructor](https://doc.cgal.org/latest/Surface_mesh_parameterization/classCGAL_1_1Surface__mesh__parameterization_1_1Square__border__parameterizer__3.html#abf28e09e7fafc69e0deadaf40ab21dc8).

**Reply by stla** [reply]
Didn't try ARAP yet. I remember there is a very slow parameterization, but I don't remember which one. Is it this one? It is so slow that I never waited for the end.

**Reply by afabri** [reply]
The constructor with the four vertices is probably more important.  
If you have an example  where ARAP is slow, it would be great if you generated an  issue with a self contained example+data so that we can have a look.  Otherwise it is just hearsay of little help.

**Reply by stla** [reply]
I found where [I claimed there's a slow parameterization](https://github.com/CGAL/cgal/discussions/7501#discussioncomment-6133881). It's  IAP, not ARAP. Will give you an example next time I try it. 

Going to see the constructor you mention now.

**Reply by stla** [reply]
Maybe it was slow because I used EPECK ?

**Reply by stla** [reply]
I see one can also specify the four corners with `Square_border_uniform_parameterizer_3`. What is the difference?

**Comment by stla** [comment]
Indeed, using ARAP with lambda=1000, the squares remain square. Here I didn't increase the number of vertices, that's why the lines are not smooth.

![arap](https://github.com/CGAL/cgal/assets/4466543/d270f1b1-ea1f-4868-bdab-c782d494bd3f)

**Comment by stla** [comment]
ARAP

![TennisBallWithCheckerboard_ARAP](https://github.com/CGAL/cgal/assets/4466543/aedf4421-dd00-40cc-a685-987145d6946a)

**Comment by stla** [comment]
I applied the DCP parameterization with the four corners method, following @afabri 's suggestion. Perfect placement of the checkerboard!

![dcpfourcorners](https://github.com/CGAL/cgal/assets/4466543/d3ae7ac3-6948-490c-be01-e75e6772892f)

**Comment by stla** [comment]
Is is possible to apply the four corners method with a parameterization which would preserve the shape of the squares, such as ARAP?

**Comment by stla** [comment]
I've found how to get an "aligned" checkerboard on the half-sphere with ARAP. It suffices to do an "aligned" checkerboard on the parameterization space:

![ARAPall](https://github.com/CGAL/cgal/assets/4466543/ac714a24-523d-4b50-a994-5b1b44074508)

**Comment by stla** [comment]
Terrific :-)

![TennisBallWithCheckerboard_ARAP_aligned](https://github.com/CGAL/cgal/assets/4466543/7ef1296d-f2fd-4ad4-8396-e56f81b121dd)
